libs/wallet-ui/src/components/interactions/flows/transaction-review/transaction-review.tsx

```diff
@@ -116,7 +126,20 @@ export const TransactionReview = ({
         title="Failed to send transaction"
         type={data.error.type}
         message={data.error.error}
-        onClose={onClose}
+        onClose={() => {
+          if (data.transaction) {
+            // this usually happens when there's a traceID mismatch
+            // eagerly mark the current tx as failure
+            dispatch({
+              type: 'UPDATE_TRANSACTION',
+              transaction: {
+                id: data.transaction?.id,
+                status: TransactionStatus.FAILURE,
```

Comment:
suggestion (blocking): if you set a failure, can you also add an error prop with some description of why this is marked as failed? when opening the transaction details, it's useful to see what happened